WordPress/Classes/ViewRelated/Reader/ReaderSubscriptionsView.swift

```diff
@@ -0,0 +1,191 @@
+            if subscriptions.isEmpty {
+                emptyStateView
+            } else {
+                main
+            }
+        }
+        .navigationTitle(Strings.title)
+    }
+
+    private var emptyStateView: some View {
+        EmptyStateView {
+            Label(Strings.title, systemImage: "doc.text.magnifyingglass")
+        } description: {
+            Text(Strings.emptyStateDetails)
+        } actions: {
+            buttonAddSubscription
+                .fixedSize(horizontal: true, vertical: /*@START_MENU_TOKEN@*/true/*@END_MENU_TOKEN@*/)
+                .padding(EdgeInsets(top: 12, leading: 8, bottom: 8, trailing: 10))
+                .background(Color(.secondarySystemBackground).opacity(0.8))
+                .clipShape(RoundedRectangle(cornerRadius: 12))
+        }
+    }
+
+    private var main: some View {
+        List {
+            if !isShowingSearchResuts {
+                buttonAddSubscription
+                    .listRowSeparator(.hidden)
+                ForEach(subscriptions, id: \.objectID, content: makeSubscriptionCell)
+                    .onDelete(perform: delete)
+            } else {
+                ForEach(searchResults, id: \.objectID, content: makeSubscriptionCell)
+                    .onDelete(perform: delete)
+            }
+        }
+        .listStyle(.plain)
+        .searchable(text: $searchText, placement: .navigationBarDrawer)
+        .onReceive(subscriptions.publisher) { _ in
+            if !searchText.isEmpty {
+                reloadSearchResults(searchText: searchText)
+            }
+        }
+        .onChange(of: searchText) {
+            reloadSearchResults(searchText: $0)
+        }
+        .toolbar {
+            EditButton()
+        }
+    }
+
+    private func makeSubscriptionCell(for site: ReaderSiteTopic) -> some View {
+        Button {
+            onSelection(site)
+        } label: {
+            ReaderSubscriptionCell(site: site, onDelete: delete)
+        }
+    }
+
+    private var buttonAddSubscription: some View {
+        Button {
+            isAddingSubscription = true
+        } label: {
+            HStack(spacing: 16) {
+                Image(systemName: "plus.circle.fill")
+                    .foregroundStyle(AppColor.brand, Color(.secondarySystemFill))
+                    .font(.largeTitle.weight(.light))
+                    .frame(width: 40)
+                    .padding(.leading, 4)
+                VStack(alignment: .leading) {
+                    Text(Strings.addSubscription)
+                        .font(.callout.weight(.medium))
+                    Text(Strings.addSubscriptionSubtitle)
+                        .font(.footnote)
+                        .foregroundStyle(.secondary)
+                }
+                .lineLimit(1)
+            }
+            .padding(.bottom, 4)
+        }
+        .buttonStyle(.plain)
+        .popover(isPresented: $isAddingSubscription) {
+            ReaderAddSubscriptionView()
+        }
```

Comment:
What do you think using an alert? If users accidentally tap any area outside of this popover, while they are typing feed url, the popover would disappear, wouldn't it? I don't think that'll happen on an alert.

Reply:
Good point. I added "Cancel" and "Add" buttons and disable interactive dismiss if input is not empty, which is all according to [HIG](https://developer.apple.com/design/human-interface-guidelines/popovers#Best-practices). Reference: "New Event" in the Calendar app.

<img width="594" alt="Screenshot 2024-09-06 at 8 56 47 AM" src="https://github.com/user-attachments/assets/dab72608-5d28-4015-8a18-06a749198160">